contracts/WETH10.sol

```diff
@@ -1,102 +1,106 @@
-pragma solidity 0.6.10;
-// Copyright (C) 2015, 2016, 2017 Dapphub / adapted by [] 2020
+pragma solidity 0.7.0;
+// Copyright (C) 2015, 2016, 2017 Dapphub / adapted by Ethereum Community 2020
 contract WETH10 {
-    string public name;
-    string public symbol;
-    uint8  public decimals;
-    bytes32 public DOMAIN_SEPARATOR;
-    bytes32 public PERMIT_TYPEHASH = keccak256("Permit(address src,address guy,uint wad,uint nonce,uint deadline)");
+    string public constant name = "Wrapped Ether";
+    string public constant symbol = "WETH";
+    uint8  public constant decimals = 18;
+    bytes32 public immutable DOMAIN_SEPARATOR;
+    bytes32 public immutable PERMIT_TYPEHASH = keccak256("Permit(address from,address spender,uint256 value,uint256 nonce,uint256 deadline)");
 
-    event  Approval(address indexed src, address indexed guy, uint wad);
-    event  Transfer(address indexed src, address indexed dst, uint wad);
-    event  Deposit(address indexed dst, uint wad);
-    event  Withdrawal(address indexed src, uint wad);
+    event  Approval(address indexed from, address indexed spender, uint256 value);
+    event  Transfer(address indexed from, address indexed to, uint256 value);
 
-    mapping (address => uint)                       public  balanceOf;
-    mapping (address => mapping (address => uint))  public  allowance;
-    mapping (address => uint)                       public  nonces;
+    mapping (address => uint256)                       public  balanceOf;
+    mapping (address => mapping (address => uint256))  public  allowance;
+    mapping (address => uint256)                       public  nonces;
     
-    constructor() public {
-        name = "Wrapped Ether";
-        symbol = "WETH";
-        decimals = 18;
-        uint chainId;
+    constructor() {
+        uint256 chainId;
         assembly {chainId := chainid()}
         DOMAIN_SEPARATOR = keccak256(
             abi.encode(
-                keccak256("EIP712Domain(string name,string version,uint chainId,address verifyingContract)"),
+                keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)"),
                 keccak256(bytes(name)),
                 keccak256(bytes("1")),
                 chainId,
                 address(this)));
     }
 
     receive() external payable {
-        deposit();
+        balanceOf[msg.sender] += msg.value;
+        emit Transfer(address(0), msg.sender, msg.value);
     }
     
-    function deposit() public payable {
+    function deposit() external payable {
         balanceOf[msg.sender] += msg.value;
-        emit Deposit(msg.sender, msg.value);
+        emit Transfer(address(0), msg.sender, msg.value);
     }
     
-    function withdraw(uint wad) external {
-        require(balanceOf[msg.sender] >= wad, "!balance");
-        balanceOf[msg.sender] -= wad;
-        (bool success, ) = msg.sender.call.value(wad)("");
+    function withdraw(uint256 value) external {
```

Comment:
My bad here, the original implementation was right. Here you are not updating the balance right.